saleor/graphql/attribute/utils.py

```diff
@@ -133,16 +136,30 @@ def _pre_save_file_value(
         if not file_url:
             return tuple()
         name = file_url.split("/")[-1]
-        value = attribute_models.AttributeValue(
-            attribute=attribute,
-            file_url=file_url,
-            name=name,
-            content_type=attr_value.content_type,
+        # don't create ne value when assignment already exists
+        value = cls._get_assigned_attribute_value_if_exists(
+            instance, attribute, attr_value.file_url
         )
-        value.slug = generate_unique_slug(value, name)  # type: ignore
-        value.save()
+        if value is None:
+            value = attribute_models.AttributeValue(
+                attribute=attribute,
+                file_url=file_url,
+                name=name,
+                content_type=attr_value.content_type,
+            )
+            value.slug = generate_unique_slug(value, name)  # type: ignore
+            value.save()
         return (value,)
 
+    @classmethod
+    def _get_assigned_attribute_value_if_exists(
+        cls, instance: T_INSTANCE, attribute: attribute_models.Attribute, file_url
+    ):
+        assignment = instance.attributes.filter(
+            assignment__attribute=attribute, values__file_url=file_url
+        ).first()
+        return None if assignment is None else assignment.values.get(file_url=file_url)
```

Comment:
`values.get()` could raise an error if the value is not present. I think this solution would be safer and should give the same result:
```
return assignment.values.filter(file_url=file_url).first()
```